server/sonar-server/src/main/java/org/sonar/server/computation/monitoring/CEQueueStatusImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.monitoring;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class CEQueueStatusImpl implements CEQueueStatus {
+  private static final long PENDING_INITIAL_VALUE = Long.MIN_VALUE;
+
+  private final AtomicLong received = new AtomicLong(0);
+  private final AtomicLong pending = new AtomicLong(PENDING_INITIAL_VALUE);
```

Comment:
Is it really needed to have AtomicLongs as counters are not updated atomically (see addReceived() and addInProgress()) ?